lib/segment/src/vector_storage/query_scorer/multi_metric_query_scorer.rs

```diff
@@ -100,23 +108,39 @@ impl<
     }
 
     fn score_stored_batch(&self, ids: &[PointOffsetType], scores: &mut [ScoreType]) {
-        debug_assert!(ids.len() <= VECTOR_READ_BATCH_SIZE);
-        debug_assert_eq!(ids.len(), scores.len());
+        let batch_size = ids.len();
+
+        debug_assert!(batch_size <= VECTOR_READ_BATCH_SIZE);
+        debug_assert_eq!(batch_size, scores.len());
 
         let mut vectors = [TypedMultiDenseVectorRef {
             flattened_vectors: &[],
-            dim: 0,
+            dim: 1,
         }; VECTOR_READ_BATCH_SIZE];
         self.vector_storage
-            .get_batch_multi(ids, &mut vectors[..ids.len()]);
-        for idx in 0..ids.len() {
+            .get_batch_multi(ids, &mut vectors[..batch_size]);
+
+        let total_read = vectors[..batch_size]
+            .iter()
+            .map(|v| v.vectors_count())
+            .sum();
+
+        self.hardware_counter
+            .vector_io_read()
+            .incr_delta(total_read);
+
+        for idx in 0..batch_size {
             scores[idx] = self.score_ref(vectors[idx]);
         }
     }
 
     fn score_internal(&self, point_a: PointOffsetType, point_b: PointOffsetType) -> ScoreType {
         let v1 = self.vector_storage.get_multi(point_a);
         let v2 = self.vector_storage.get_multi(point_b);
+        self.hardware_counter
+            .cpu_counter()
+            .incr_delta(v1.vectors_count() + v2.vectors_count());
+
```

Comment:
Do we not need to measure vector_io here and in other places?

Reply:
yes, this was a blunder